sql/migrations/state/0022_malfeasance.sql

```diff
@@ -0,0 +1,20 @@
+-- adds new table for v2 malfeasance proofs
+-- TODO(mafa): in the future add a migration to convert old malfeasance proofs to the new format
+--    and then remove proof, received from the old table
+
+CREATE TABLE malfeasance
+(
+    pubkey      CHAR(32) PRIMARY KEY,
+    received    INT NOT NULL, -- unix timestamp
+
+    -- if the following field is not null, then domain and proof are null
```

Comment:
if there's a foreign key constraint, doesn't it mean that `married_to` becomes always mandatory? (aka: `not null`)

Reply:
No, there are even tests for that in the `mafleasance_test.go` file. `married_to` can be null `NULL`, but if it isn't its value must also be present in another row in the `pubkey` column.